datafusion/common/src/dfschema.rs

```diff
@@ -297,6 +297,19 @@ impl DFSchema {
 
     /// Modify this schema by appending the fields from the supplied schema, ignoring any
     /// duplicate fields.
+    ///
+    /// ## Merge Precedence
+    ///
+    /// **Schema-level metadata**: Metadata from both schemas is merged.
+    /// If both schemas have the same metadata key, the value from the second `other_schema` parameter takes precedence.
+    ///
+    /// **Field-level merging**: Only non-duplicate fields are added. This means that the
+    /// first `self` fields will always take precedence over the second `other_schema` fields.
+    /// Duplicate field detection is based on:
+    /// - For qualified fields: both qualifier and field name must match
+    /// - For unqualified fields: only field name needs to match
+    ///
+    /// Note: the merging operation prefers the first `self` fields, and the second `other_schema` metadata.
```

Comment:
```suggestion
    /// Take note how the precedence for fields & metadata merging differs; merging prefers fields from `self` but prefers metadata from`other_schema`.
```

Maybe highlighting in this way might make it clearer? To explicitly call out there is a difference (rather than the user reading it and realizing there is supposed to be a difference)